core/src/test/kotlin/com/malinskiy/marathon/execution/queue/QueueActorSpek.kt

```diff
@@ -52,41 +58,32 @@ class QueueActorSpek : Spek({
                         job = job
                 )
             }
-
-            describe("requesting batch and handling completed batch") {
-                beforeEachTest {
-                    runBlocking {
-                        actor.send(QueueMessage.RequestBatch(TEST_DEVICE_INFO))
-                        poolChannel.receive()
-                    }
+            val captor = argumentCaptor<TestResult>()
+            val results = createBatchResult(uncompleted = listOf(
+                    createTestResult(TEST_1, TestStatus.FAILURE)
+            ))
+
+            it("should have empty queue") {
+                val isEmptyDeferred = CompletableDeferred<Boolean>()
+                runBlocking {
+                    actor.send(QueueMessage.RequestBatch(TEST_DEVICE_INFO))
+                    poolChannel.receive()
+                    actor.send(QueueMessage.Completed(TEST_DEVICE_INFO, results))
+                    actor.send(QueueMessage.IsEmpty(isEmptyDeferred))
+                    isEmptyDeferred.await() shouldBe true
                 }
+            }
 
-                describe("received uncompleted test once") {
-                    beforeEachTest {
-                        val results = createBatchResult(uncompleted = listOf(
-                                createTestResult(TEST_1, TestStatus.FAILURE)
-                        ))
-                        runBlocking {
-                            actor.send(QueueMessage.Completed(TEST_DEVICE_INFO, results))
-                        }
-                    }
-
-                    it("should have empty queue") {
-                        val isEmptyDeferred = CompletableDeferred<Boolean>()
-                        runBlocking {
-                            actor.send(QueueMessage.IsEmpty(isEmptyDeferred))
-                            isEmptyDeferred.await() shouldBe true
-                        }
-                    }
-
-                    it("should report test as failed") {
-                        runBlocking {
-                            val captor = argumentCaptor<TestResult>()
-                            verify(analytics).trackTestFinished(any(), any(), captor.capture())
-                            captor.firstValue.test shouldBe TEST_1
-                            captor.firstValue.status shouldBe TestStatus.FAILURE
-                        }
-                    }
+            it("should report test as failed") {
+                declareMock<Track> {}
+                val track: Track by inject()
+                runBlocking {
+                    actor.send(QueueMessage.RequestBatch(TEST_DEVICE_INFO))
+                    poolChannel.receive()
+                    actor.send(QueueMessage.Completed(TEST_DEVICE_INFO, results))
+                    verify(track).test(any(), any(), captor.capture(), any())
+                    captor.firstValue.test shouldBe TEST_1
```

Comment:
Do we really need to have this whitebox tests? Why not test it as a black box?

Reply:
I don't think having more tests is a bad thing. Removing tests is fine when they're not checking anything